File: Invelop/Invelop.Project.Services/Person/PersonContactsService.cs
using Invelop.Project.Core.Models;
using Invelop.Project.Repository.Person;

namespace Invelop.Project.Services.Person
{
    public class PersonContactsService : IPersonContactsService
    {
        private readonly IPersonContactsRepository _personContactsRepository;

        public PersonContactsService(IPersonContactsRepository personContactsRepository)
        {
            _personContactsRepository = personContactsRepository;
        }

        public async Task<IEnumerable<PersonContacts>> GetAll()
        {
            return await _personContactsRepository.GetAll();
        }

        public async Task<PersonContacts?> Get(long Id)
        {
            return await _personContactsRepository.Get(Id);
        }

        public async Task<long> Insert(PersonContacts personContacts)
        {
            return await _personContactsRepository.Insert(personContacts);
        }

        public async Task<bool> Update(PersonContacts personContacts)
        {
            return await _personContactsRepository.Update(personContacts);
        }

        public async Task<bool> Delete(long Id)
        {
            return await _personContactsRepository.Delete(Id);
        }
    }
}

File: Invelop/Invelop.Project/ClientApp/src/app/person-contacts/shared/services/person-contacts.service.ts
import { HttpClient } from '@angular/common/http';
import { Inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { PersonContacts } from '../models/person-contacts.model';

@Injectable({
  providedIn: 'root'
})
export class PersonContactsService {
  private readonly baseUrl: string;

  constructor(private http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
    this.baseUrl = baseUrl;
  }

  public GetAll(): Observable<PersonContacts[]> {
    return this.http.get<PersonContacts[]>(this.baseUrl + 'personcontacts');
  }

  public Delete(Id: number): Observable<any> {
    return this.http.delete(this.baseUrl + 'personcontacts/' + Id);
  }

  public Get(Id: number): Observable<PersonContacts> {
    return this.http.get<PersonContacts>(this.baseUrl + 'personcontacts/' + Id);
  }

  public Insert(personContacts: PersonContacts): Observable<any> {
    return this.http.post<any>(this.baseUrl + 'personcontacts', personContacts);
  }

  public Update(personContacts: PersonContacts): Observable<any> {
    return this.http.put<any>(this.baseUrl + 'personcontacts', personContacts);
  }  
}
